modules/text_generation.py

```diff
@@ -314,24 +313,30 @@ def generate_with_streaming(**kwargs):
 def generate_reply_custom(question, original_question, seed, state, stopping_strings=None, is_chat=False):
     seed = set_manual_seed(state['seed'])
 
+    # Use 'encode' to truncate the prompt
+    truncated_prompt_ndarray = encode(question, truncation_length=get_max_prompt_length(state))
+
+    # Convert the truncated prompt back to a string (to avoid modifying 'generate' and 'generate_with_streaming')
+    truncated_prompt_string = shared.model.decode(truncated_prompt_ndarray[0])
+
     t0 = time.time()
     reply = ''
     try:
         if not is_chat:
             yield ''
 
         if not state['stream']:
-            reply = shared.model.generate(question, state)
+            reply = shared.model.generate(truncated_prompt_string, state)
             yield reply
         else:
-            for reply in shared.model.generate_with_streaming(question, state):
+            for reply in shared.model.generate_with_streaming(truncated_prompt_string, state):
                 yield reply
 
     except Exception:
         traceback.print_exc()
     finally:
         t1 = time.time()
-        original_tokens = len(encode(original_question)[0])
-        new_tokens = len(encode(original_question + reply)[0]) - original_tokens
+        original_tokens = len(truncated_prompt_ndarray[0])
```

Comment:
Have "context" report the actual context used by the model, not the number of tokens of the text in the UI.